### PR TITLE
Improved error logging for win_task.create_task_from_xml

### PR DESCRIPTION
**Pull request by Lesvek**
### What does this PR do?
Instead of only returning True or False when create_task_from_xml fail to add the scheduled task, the actual error message is returned to the user. If the error is unknown, the hex error code is returned. That way, it's easier to search for as this is the format Microsoft is using in their documentation.

### Previous Behavior
Previously, when there was an error, the module only returned False, only logging the error message in the debug logs.

### New Behavior
A proper exception containing the error message is now returned to the user when there is an error. New code will have to catch the exception.

### Tests written?
I added 2 new tests.

### Commits signed with GPG?
The commit was not signe with GPG.

### CODE REVIEW

**Review by twangboy — APPROVED** [review]

File: salt/modules/win_task.py [hunk 1]
```diff
@@ -621,21 +622,38 @@ def create_task_from_xml(name,
...
-            failure_code = fc[exc[5]]
+            failure_code = fc[error_code]
         except KeyError:
-            failure_code = 'Unknown Failure: {0}'.format(error)
-
-        log.debug('Failed to create task: %s', failure_code)
+            failure_code = 'Unknown Failure: {0}'.format(error_code)
+        finally:
+            log.debug('Failed to create task: %s', failure_code)
+        return failure_code
```
Comment (twangboy) [inline salt/modules/win_task.py]: We should probably raise a CommandExecutionError here instead. If the Return type is Bool, it should always be Bool. It shouldn't return 2 different types.

**Review by twangboy — CHANGES_REQUESTED** [review]

File: salt/modules/win_task.py [hunk 2]
```diff
@@ -566,8 +566,8 @@ def create_task_from_xml(name,
         the task to run whether the user is logged in or not, but is currently
         not working.
 
-    :return: True if successful, False if unsuccessful
-    :rtype: bool
+    :return: True if successful, False if unsuccessful, A string with the error message if there is an error
+    :rtype: bool or str
```
Comment (twangboy) [inline salt/modules/win_task.py]: Add a `:raises: CommandExecutionError` for completenes

File: salt/modules/win_task.py (hunk 1 above)
Comment (Lesvek) [reply salt/modules/win_task.py]: Oh, nice. I didn't know it existed.

**Review by twangboy — CHANGES_REQUESTED** [review]

File: tests/integration/modules/test_win_task.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+from tests.support.case import ModuleCase
+from tests.support.unit import skipIf
+from tests.support.helpers import destructiveTest
+
+import salt.utils.platform
```
[the remark this answers is not in the document]
Comment (twangboy) [reply tests/integration/modules/test_win_task.py]: And you need two blank lines between the imports and the class (lint)

File: tests/integration/modules/test_win_task.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
```
Comment (twangboy) [inline tests/integration/modules/test_win_task.py]: You probably need to add this to fix a lint error:
```
from __future__ import absolute_import, unicode_literals
```

File: tests/integration/modules/test_win_task.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        all_tasks = self.run_function('task.list_tasks')
+        self.assertIn('foo', all_tasks)
+
+    @destructiveTest
+    def test_adding_task_with_invalid_xml(self):
+        '''
+        Test adding a task using xml
+        '''
+        xml_text = r"""<Malformed"""
+        with self.assertRaises(salt.exceptions.CommandExecutionError):
```
Comment (twangboy) [inline tests/integration/modules/test_win_task.py]:
```
with self.assertRaises(CommandExecutionError):
```

**Review by twangboy — APPROVED** [review]

**Comment by twangboy** [comment]
@Lesvek Thanks for getting all those things fixed and thanks for the contribution! Looks good now.

**Pull request merged**